client/extensions/woocommerce/state/sites/orders/notes/reducer.js

```diff
@@ -34,6 +37,26 @@ export function isLoading( state = {}, action ) {
 }
 
 /**
+ * Returns the updated order saving state after an action has been
+ * dispatched. This reflects a mapping of order ID to a boolean,
+ * indicating whether there is a save in progress.
+ *
+ * @param  {Object} state  Current state
+ * @param  {Object} action Action payload
+ * @return {Object}        Updated state
+ */
+export function isSaving( state = {}, action ) {
+	switch ( action.type ) {
+		case WOOCOMMERCE_ORDER_NOTE_CREATE:
+		case WOOCOMMERCE_ORDER_NOTE_CREATE_FAILURE:
+		case WOOCOMMERCE_ORDER_NOTE_CREATE_SUCCESS:
+			return Object.assign( {}, state, { [ action.orderId ]: WOOCOMMERCE_ORDER_NOTE_CREATE === action.type } );
```

Comment:
Habit :) and isLoading (in this file) uses Object.assign (along with the other reducers I've written). If we have standardized on the spread operator instead I can make another PR to change those next week.